3.5-windowsservercore-1709/build/Dockerfile

```diff
@@ -23,6 +23,15 @@ RUN Invoke-WebRequest -UseBasicParsing https://download.visualstudio.microsoft.c
     Remove-Item -Force -Recurse ${Env:TEMP}\*; `
     Remove-Item -Force -Recurse \"${Env:ProgramData}\Package Cache\"
 
+# ngen assemblies queued by VS installers - must be done in cmd shell to avoid access issues
+SHELL ["cmd", "/S", "/C"]
+RUN set COMPLUS_NGenProtectedProcess_FeatureEnabled=0 `
+    && \Windows\Microsoft.NET\Framework64\v4.0.30319\ngen update `
+    # Workaround VS installer/ngen issue with VSWebLauncher
```

Comment:
Comments in a multi-line run command causes - `Empty continuation lines will become errors in a future release.` Per https://github.com/moby/moby/pull/35004 should be resolved in 17.10 or newer.